pkg/pinger/pinger.go

```diff
@@ -154,8 +173,8 @@ func (p *pingerInfo) doPing() error {
 
 		// Pinger will mark a connection as an error if the packet loss reaches the threshold
 		if pinger.PacketsSent-pinger.PacketsRecv > int(p.maxPacketLossCount) {
-			p.Lock()
-			defer p.Unlock()
+			p.Mutex.Lock()
```

Comment:
Don't need to reference `Mutex`.

```suggestion
			p.Lock()
```